gh-gl-sync/SpackCIBridge.py

```diff
@@ -400,10 +467,19 @@ def sync(self):
             # Setup the local repo with two remotes.
             self.setup_git_repo()
 
+            # Find the currently running develop pipeline, if any, and get the sha
+            currently_running_develop_sha = None
+            develop_pipelines = self.get_pipelines_for_branch("develop", self.time_threshold_dev)
+            for dev_sha, dev_pipeline in develop_pipelines.items():
+                if dev_pipeline['status'] == "running":
+                    currently_running_develop_sha = dev_sha
```

Comment:
```suggestion
                    currently_running_develop_sha = dev_sha
                    break
```

Reply:
There's no guarantee about the ordering of the returned pipelines, is there?  Is there any particular reason to pick the first one rather than the last?  Note that in our case at least, there's only ever going to be a single develop pipeline running anyway.

Reply:
As far as I know, you're right: there's no guarantee what order GitLab's API will return its results in.

I guess if we wanted to be very thorough we would pull /all/ the running develop pipelines and check against all of them. What do you think, probably overkill?

Reply:
To me it seems like overkill, considering our target use case involves pushing to a gitlab where "cancel redundant pending pipelines" is going to have to remain enabled forever.  Does the other project using this script not enable that feature?

Reply:
As far as I'm aware Spack is the only project using this script that auto-cancels redundant pipelines. So I agree, probably overkill. Let's just grab the first (only) running dev pipeline & break out of the loop?